src/utils/builder.ts

```diff
@@ -0,0 +1,310 @@
+import { Pathway, Criteria, State, Transition, Action, GuidanceState } from 'pathways-model';
+import shortid from 'shortid';
+import { MedicationRequest, ServiceRequest } from 'fhir-objects';
+
+export function createNewPathway(name: string, description?: string): Pathway {
+  return {
+    name: name,
+    description: description ?? '',
+    library: '',
+    criteria: [],
+    states: {
+      Start: {
+        key: 'Start',
+        label: 'Start',
+        transitions: []
+      }
+    }
+  };
+}
+
+export function exportPathway(pathway: Pathway): string {
+  // Strip id from each criteria
+  pathway.criteria.forEach((criteria: Criteria) => delete criteria.id);
+
+  // Strip key from each state
+  Object.keys(pathway.states).forEach((stateName: string) => {
+    const state = pathway.states[stateName];
+    delete state.key;
+
+    // Strip id from each state.transition
+    state.transitions.forEach((transition: Transition) => delete transition.id);
+
+    // Strip id from each state.action
+    if ('action' in state) {
+      (state as GuidanceState).action.forEach((action: Action) => delete action.id);
+    }
+  });
+
+  return JSON.stringify(pathway, undefined, 2);
+}
+
+export function addLibrary(pathway: Pathway, library: string): void {
+  pathway.library = library;
+}
+
+// TODO: possibly add more criteria methods
+export function addCriteria(
+  pathway: Pathway,
+  elementName: string,
+  expected: string,
+  cql: string
+): string {
+  const id = shortid.generate();
+  const criteria: Criteria = {
+    id: id,
+    elementName: elementName,
+    expected: expected,
+    cql: cql
+  };
+  pathway.criteria.push(criteria);
+
+  return id;
+}
+
+export function addNavigationalElm(pathway: Pathway, elm: object): void {
```

Comment:
Minor point but we should standardize on naming convention. Generally `addX` means adding to a list and `setX` sets a value so this and a few other places should be `setX`

Reply:
Update so only methods adding to a list are called `addX` and those which are updating a value to be `setX`